File: android/app/src/main/java/com/baatcheet/app/data/remote/dto/AuthDto.kt
package com.baatcheet.app.data.remote.dto

import com.baatcheet.app.domain.model.User
import com.google.gson.annotations.SerializedName

/**
 * Data Transfer Objects for Authentication API
 * 
 * These DTOs represent the exact structure of API responses.
 * They are mapped to domain models before being used in the app.
 */

// Response wrapper
data class AuthResponseDto(
    val success: Boolean,
    val data: AuthDataDto?,
    val error: String?
)

// Auth data
data class AuthDataDto(
    val user: UserDto?,
    val token: String?,
    val status: String?,
    val message: String?,
    val email: String?
)

// User DTO
data class UserDto(
    val id: String,
    val email: String,
    @SerializedName("firstName") val firstName: String?,
    @SerializedName("lastName") val lastName: String?,
    val avatar: String?,
    val role: String?,
    val tier: String?
) {
    /**
     * Map DTO to domain model
     */
    fun toDomain(): User = User(
        id = id,
        email = email,
        firstName = firstName,
        lastName = lastName,
        avatar = avatar,
        role = role,
        tier = tier
    )
}

// Request DTOs
data class SignInRequestDto(
    val email: String,
    val password: String
)

data class SignUpRequestDto(
    val email: String,
    val password: String,
    val firstName: String? = null,
    val lastName: String? = null
)

data class VerifyEmailRequestDto(
    val email: String,
    val code: String
)

data class ResendCodeRequestDto(
    val email: String
)

File: ios/iOSBaseProject/Data/Remote/DTOs/AuthDTO.swift
//
//  AuthDTO.swift
//  BaatCheet
//
//  Data Transfer Objects for Authentication API
//

import Foundation

// MARK: - Response DTOs

/// API Response wrapper
struct AuthResponseDTO: Codable {
    let success: Bool
    let data: AuthDataDTO?
    let error: String?
}

/// Auth data from API
struct AuthDataDTO: Codable {
    let user: UserDTO?
    let token: String?
    let status: String?
    let message: String?
    let email: String?
}

/// User data from API
struct UserDTO: Codable {
    let id: String
    let email: String
    let firstName: String?
    let lastName: String?
    let avatar: String?
    let role: String?
    let tier: String?
    
    /// Map DTO to domain model
    func toDomain() -> User {
        User(
            id: id,
            email: email,
            firstName: firstName,
            lastName: lastName,
            avatar: avatar,
            role: role,
            tier: tier
        )
    }
}

// MARK: - Request DTOs

/// Sign in request body
struct SignInRequestDTO: Codable {
    let email: String
    let password: String
}

/// Sign up request body
struct SignUpRequestDTO: Codable {
    let email: String
    let password: String
    let firstName: String?
    let lastName: String?
}

/// Email verification request body
struct VerifyEmailRequestDTO: Codable {
    let email: String
    let code: String
}

/// Resend code request body
struct ResendCodeRequestDTO: Codable {
    let email: String
}

// MARK: - Me Response DTO

struct MeResponseDTO: Codable {
    let success: Bool
    let data: MeDataDTO?
}

struct MeDataDTO: Codable {
    let user: UserDTO
}
